### ISSUE TITLE
Deprecate InitFrom in collection create

### ISSUE THREAD

**Issue by generall** [qdrant/qdrant#2911]
**Is your feature request related to a problem? Please describe.**

InitFrom was used to solve some limitation of the API, like inability to change HNSW params, which are no longer applicable.
Additionally, InitFrom provides no guarantees in data consistency if the original collection is updated, so in practice, it is only really usable in development env under very specific circumstances.

The stability of this function is also questionable

**Describe the solution you'd like**

Mark parameter as deprecated, print WARN if it is used, and eventually remove the functionality.

**Comment by timvisee** [comment]
We still point users to use this for some operations, for example, to update the replica factor. I know all data can be inserted again, but InitFrom is still very convenient.

It would be great if we could invest some time in making the number of shards or the replica factor configurable at runtime as well. I guess that would eliminate the need for this API other than just cloning a collection.

I wonder if we could (easily) guarantee consistency while updates are happening by replaying the WAL (from init start) once the other collection is initialized.

**Comment by generall** [comment]
I don't think that the changing replication factor is a good enough reason to use this feature, probably you meant changing the number of shards.
